File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    limit_x, limit_y = (board[0]-1)//2, (board[1]-1)//2
    
    def is_valid(x,y):
        return -limit_x<=x<=limit_x and -limit_y<=y<=limit_y
    
    x, y = 0, 0
    
    for key in keyinput:
        if key == "left":
            if is_valid(x-1,y):
                x -= 1
        if key == "right":
            if is_valid(x+1,y):
                x += 1
        if key == "up":
            if is_valid(x,y+1):
                y+=1
        if key == "down":
            if is_valid(x,y-1):
                y -= 1
    
    return [x,y]

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[]{0,0};
        int a = board[0]/2;
        int b = board[1]/2;
        for (String key : keyinput){
            if (key.equals("left")) {
                if (answer[0]-1>=-a) answer[0] -= 1;
            } else if (key.equals("right")) {
                if (answer[0]+1<=a) answer[0] += 1;
            } else if (key.equals("up")) {
                if (answer[1]+1<=b) answer[1] += 1;
            } else{
                if (answer[1]-1>=-b) answer[1] -= 1;
            }
        }
        return answer;
    }
}